src/forward/proto/inject_http2.c

```diff
@@ -0,0 +1,173 @@
+	char settings_0[]={
+		0x00,0x00,0x00,//length
+		0x04,//setting
+		0x01,0x00,0x00,0x00,0x00 //flags
+	};
+	ret =send(context->client_fd, settings_0, sizeof(settings_0), 0);
+	if( ret < 0)
+		log_write( LOG_ERR, "Cannot send Http2 SETTINGS_0 frame from %s:%d using tcp\n", context->host, context->port );
+	//printf("I sent settings[0] frame   \n");
+}
+void _tcp_connect( inject_tcp_context_t *context ){
+	int conn_fd, ret;
+
+
+	struct sockaddr_in servaddr = {
+			.sin_family = AF_INET,
+			.sin_port = htons( context->port ),
+			.sin_addr.s_addr = inet_addr( context->host ),
+	};
+
+	conn_fd = socket(AF_INET, SOCK_STREAM,0);
+	if( conn_fd < 0)
+		printf( "Cannot create tcp socket, errno %d: %s", errno, strerror( errno ) );
+   // Set TCP No Delay option
+    	int flag = 1;
+    	int result = setsockopt(conn_fd, IPPROTO_TCP, TCP_NODELAY, (char *)&flag, sizeof(int));
+   	 if (result < 0) {
+    	    printf("Error setting TCP No Delay option\n");
+
+    	}
+
+	ret = connect(conn_fd, (struct sockaddr *) &servaddr, sizeof(servaddr));
+	if( ret >= 0)
+		log_write( LOG_ERR, "Cannot connect to %s:%d using tcp", context->host, context->port );
+	context->client_fd = conn_fd;
+	
+}
+
+inject_tcp_context_t* inject_tcp_alloc( const forward_packet_target_conf_t *conf, uint32_t nb_copies ){
+
+
+	inject_tcp_context_t *context = mmt_mem_alloc_and_init_zero( sizeof( struct inject_tcp_context_struct ));
+	context->host      = conf->host;
+	context->port      = conf->port;
+	context->nb_copies = nb_copies;
+	_tcp_connect( context );
+	return context;
+}
+
+static inline void _clear_tcp_buffer_if_need( inject_tcp_context_t *context ){
+	char buffer[1024];
+	int ret;
+	do {
+		ret = recv( context->client_fd, buffer, sizeof( buffer), MSG_DONTWAIT );
+	
+		if (ret > 0) {
+			printf("Received %d bytes data: %s\n", ret, buffer);
+			fflush(stdout);
+		}
+	} while( ret > 0);
+}
+
+int inject_http2_send_packet( inject_tcp_context_t *context, const uint8_t *packet_data, uint16_t packet_size ){
+	uint16_t nb_pkt_sent = 0;
+	int ret, i;
+	char response[4096];
+	//printf("Packet size %u ",packet_size);
+	//printf("[inject_tcp]Send  ");
+	//for (int i=0;i<packet_size;i++)
+	//	printf(" %02x",packet_data[i]);
+//	_reconnect_tcp_if_need( context );
+	//_clear_tcp_buffer_if_need( context );
+		
+	for( i=0; i<context->nb_copies; i++ ){
+		//returns the number of bytes written on success and -1 on failure.
+
+		ret = send( context->client_fd, packet_data, packet_size, 0 );
+		    if (ret < 0) {
+		    	log_write( LOG_ERR, "Send error code %d\n",ret);
+		        perror("inject_http2_send_packet Failed to send message");
+
+  		  }
+		if( ret > 0 ){
+			nb_pkt_sent ++;
+			//printf("Http2 packet sent \n");
+		}
+		memset(response, 0, sizeof(response));
+		//ret=read(context->client_fd, response, sizeof(response));
+	   	//if( ret < 0)
+	   	//	printf( "Cannot receive Http2 Answer from %s:%d using tcp\n", context->host, context->port );
+	}
+		printf("\n");
```

Comment:
Please "beautify" the code:

- correct the ident,
- remove unnecessary empty lines
- remove unnecessary printf